### DISCUSSION TITLE
MultipartForm upload with non ASCII file name

### DISCUSSION THREAD

**Discussion by tim-93** [quarkusio/quarkus#44362] [Community]
quarkus:3.15.1
jdk:21

Using the ASCII file name, the file content is normal, and the non-ASCII file name, the file content is null, I don't understand why

`
@POST
  @Consumes(MediaType.MULTIPART_FORM_DATA)
  @Produces(MediaType.APPLICATION_JSON)
  public Response test(@MultipartForm UploadFile uploadFile) {
.....
}

@Data
public class UploadFile {

  @FormParam("file")
  @PartType(MediaType.APPLICATION_OCTET_STREAM)
  public InputStream file;
}
`

**Comment by gsmet** [comment]
I think a small Maven reproducer would be helpful.

**Reply by tim-93** [reply]
Is it a problem with the Maven version? The Maven version I use is 3.9.3

**Reply by gsmet** [reply]
No no, I'm just asking for a reproducer so that we can check what's going on. I added the Maven precision as it's sometimes harder for us to deal with Gradle reproducers.

**Reply by tim-93** [reply]
Hi，Please refer to me uploaded, thank you!
[quakrus-demo.zip](https://github.com/user-attachments/files/17672381/quakrus-demo.zip)

**Reply by gsmet** [reply]
@dontgm can you clarify how you trigger the issue with your demo so that I can get the exact same problem you are experiencing?

Thanks!

**Reply by tim-93** [reply]
You need to use a development tool, such as IntelliJ IDEA, import the demo project, and after starting it, use the interface to call the tool, such as Postman, to upload the file request interface. If the file name is ASCII, the interface returns: File is not null!, which is correct. Otherwise, the interface returns: File is null!, which is incorrect.

![image](https://github.com/user-attachments/assets/93fec34f-47ad-4ab5-88cc-f6377801edf9)
![image](https://github.com/user-attachments/assets/7332b2a2-537d-4768-8346-f9bd47385e47)
![image](https://github.com/user-attachments/assets/42d01029-96ab-42b0-bfaf-bfe48c47269b)

**Reply by harchiki** [reply]
Also, one more example of this problem is this issue on stackoverflow : [https://stackoverflow.com/questions/78828849/multipartformdatainput-upload-with-non-ascii-file-name-quarkus](url)

i think we all have same problem

**Comment by harchiki** [comment]
i have same problem. i think somehow fileName* couldnt be parsed in Content-Disposition
![Screenshot_2](https://github.com/user-attachments/assets/9e5a05a8-dcad-4951-811e-47c8d2048dc9)

i think that causes file body not being find by fileName (it seems encoded properly but its body is set with uncorrect fileName as key in Map)
![Screenshot_3](https://github.com/user-attachments/assets/18e2f00c-4928-4efd-84f7-0501b05e1ca7)

i couldnt force it to encode properly whatever i ve tried

**Reply by tim-93** [reply]
I have tried to encode the file name as much as possible, but I didn't expect to find a better solution...

**Reply by harchiki** [reply]
I have no idea about its solution. I ve debugged its library. Appearantly, it can parse the name in content-disposition. On filename*=UTF-8' 'non-ascii-file-name.pfd, something happens weard. Let's see the response by the reproducer. Otherwise, i will try to consider getting it around somehow
